icu4c/source/test/fuzzer/date_time_pattern_generator_fuzzer.cpp

```diff
@@ -0,0 +1,53 @@
+// © 2019 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include <cstring>
+
+#include "fuzzer_utils.h"
+#include "unicode/dtptngen.h"
+#include "unicode/localpointer.h"
+#include "unicode/locid.h"
+
+IcuEnvironment* env = new IcuEnvironment();
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  UErrorCode status = U_ZERO_ERROR;
+
+  uint16_t rnd16;
+
+  if (size < 2 + sizeof(rnd16))
+    return 0;
+
+  std::memcpy(&rnd16, data, sizeof(rnd16));
+  size -= sizeof(rnd16);
+  data += sizeof(rnd16);
+  const icu::Locale& locale = GetRandomLocale(rnd16);
+
+  std::unique_ptr<char16_t[]> fuzzbuff(new char16_t[size/2]);
+  std::memcpy(fuzzbuff.get(), data, (size/2)*2);
+  icu::UnicodeString fuzzstr(false, fuzzbuff.get(), size/2);
+
+  icu::LocalPointer<icu::DateTimePatternGenerator > gen(
+      icu::DateTimePatternGenerator::createInstance(locale, status), status);
+  if (U_FAILURE(status))
+    return 0;
+
+  status = U_ZERO_ERROR;
+  gen->getSkeleton(fuzzstr, status);
+
+  status = U_ZERO_ERROR;
+  gen->getBaseSkeleton(fuzzstr, status);
+
+  status = U_ZERO_ERROR;
+  gen->getBaseSkeleton(fuzzstr, status);
+
+  status = U_ZERO_ERROR;
+  gen->getPatternForSkeleton(fuzzstr);
```

Comment:
The thing you _really_ want to be calling (the primary API function for `DateTimePatternGenerator`) is `getBestPattern()`.  I think it's good to call all these other APIs too, but that's the one where the real meat is.

And as in some of these other fuzz tests, if you're just passing a random string here, most of the time you're going to get an error or nonsensical output.  Given the purpose of fuzz testing, this is fine, but there might also be value in additional tests that know more about the expected syntax and play around with it in interesting ways.